packages/flutterfire_cli/lib/src/firebase/firebase_apple_setup.dart

```diff
@@ -0,0 +1,654 @@
+    final applePlatform = flutterConfig[kPlatforms] as Map;
+    final appleConfig =
+        applePlatform[platform.toLowerCase() == 'ios' ? kIos : kMacos] as Map;
+
+    final configurationMaps = appleConfig[configuration] as Map?;
+
+    Map? configurationMap;
+    // For "build configuration" or "target" we need to create a nested map if it does not exist
+    if (ProjectConfiguration.target == projectConfiguration ||
+        ProjectConfiguration.buildConfiguration == projectConfiguration) {
+      if (configurationMaps?[targetOrBuildConfiguration] == null) {
+        // ignore: implicit_dynamic_map_literal
+        configurationMaps?[targetOrBuildConfiguration] = {};
+        configurationMap =
+            configurationMaps?[targetOrBuildConfiguration] as Map;
+      }
+    } else {
+      // Only a single map in "default" configuration.
+      configurationMap = configurationMaps;
+    }
+
+    configurationMap?[kProjectId] = projectId;
+    configurationMap?[kAppId] = appId;
+    configurationMap?[kUploadDebugSymbols] = debugSymbolScript;
+    configurationMap?[kServiceFileOutput] = relativePathFromProject;
+
+    final mapJson = json.encode(map);
+
+    file.writeAsStringSync(mapJson);
+  }
+
+  bool _shouldRunUploadDebugSymbolScript(
+    bool? generateDebugSymbolScript,
+    Logger logger,
+  ) {
+    if (generateDebugSymbolScript != null) {
+      return generateDebugSymbolScript;
+    } else {
+      // Unspecified, so we prompt
+      final addSymbolScript = promptBool(
+        "Do you want an '$debugSymbolScriptName' adding to the build phases of your $platform project?",
+      );
+
+      if (addSymbolScript == false) {
+        logger.stdout(
+          logSkippingDebugSymbolScript,
+        );
+      }
+      return addSymbolScript;
+    }
+  }
+
+  Future<void> _updateFirebaseJsonAndDebugSymbolScript(
+    String pathToServiceFile,
+    ProjectConfiguration projectConfiguration,
+    String targetOrBuildConfiguration,
+  ) async {
+    final runDebugSymbolScript = _shouldRunUploadDebugSymbolScript(
+      generateDebugSymbolScript,
+      logger,
+    );
+
+    if (runDebugSymbolScript) {
+      await _addFlutterFireDebugSymbolsScript(
+        xcodeProjFilePath,
+        logger,
+        projectConfiguration,
+      );
+    }
+
+    await _updateFirebaseJsonFile(
+      flutterApp!,
+      platformOptions.appId,
+      platformOptions.projectId,
+      runDebugSymbolScript,
+      targetOrBuildConfiguration,
+      pathToServiceFile,
+      projectConfiguration,
+    );
+  }
+
+  String? _addPathToExecutablesForBuildPhaseScripts() {
+    final envVars = Platform.environment;
+    final paths = envVars['PATH'];
+    if (paths != null) {
+      final array = paths.split(':');
+
+      final pathsToAddToScript = array.where((path) {
+        if (path.contains('dart-sdk') ||
+            path.contains('flutter') ||
+            path.contains('.pub-cache')) {
+          return true;
+        }
+        return false;
```

Comment:
Could we add some code comments that explain what this is doing, maybe some input & output example comments